core/src/main/java/hudson/slaves/SlaveComputer.java

```diff
@@ -482,7 +481,15 @@ public void onClosed(Channel c, IOException cause) {
                     taskListener.getLogger().println("Connection terminated");
                 }
                 closeChannel();
-                launcher.afterDisconnect(SlaveComputer.this, taskListener);
+                try {
+                    launcher.afterDisconnect(SlaveComputer.this, taskListener);
+                } catch (Throwable t) {
+                    LogRecord lr = new LogRecord(Level.SEVERE,
+                            "Launcher {0}'s afterDisconnect method propagated an exception when {1}'s connection was closed: {2}");
+                    lr.setThrown(t);
+                    lr.setParameters(new Object[]{launcher, SlaveComputer.this, t.getMessage()});
```

Comment:
@jtnord you assume that people are not supressing stack traces in their logs

@olivergondza maybe... but that would need to be in core and then you'd need to wait for core to provide it... in which case I'd much rather just switch all of core to slf4j rather than re-invent the wheel